libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -628,6 +632,35 @@ std::vector<std::shared_ptr<Vote>> VoteManager::getRewardVotes() {
   return reward_votes;
 }
 
+std::vector<std::shared_ptr<Vote>> VoteManager::getRewardVotes(const std::vector<vote_hash_t>& vote_hashes) {
+  std::vector<std::shared_ptr<Vote>> reward_votes;
+
+  std::shared_lock lock(reward_votes_mutex_);
+  for (auto vote_hash : vote_hashes) {
+    auto it = reward_votes_.find(vote_hash);
+    if (it != reward_votes_.end()) {
+      reward_votes.emplace_back(it->second);
+    } else {
+      LOG(log_dg_) << "Missing reward vote: " << vote_hash;
```

Comment:
I would return immediately with empty array. No need to continue

Reply:
Fixed